libretroshare/src/rsserver/p3peers.cc

```diff
@@ -1104,10 +1111,139 @@ bool p3Peers::GetPGPBase64StringAndCheckSum(	const RsPgpId& gpg_id,
 	return true ;
 }
 
+bool p3Peers::getShortInvite(
+        std::string& invite, const RsPeerId& pSslId, bool formatRadix,
+        bool includeFingerprint, bool bareBones )
+{
+	const RsPeerId& sslId(pSslId.isNull() ? getOwnId() : pSslId);
+
+	RsPeerDetails dts;
+	if (!getPeerDetails(sslId, dts)) return false;
+
+	RsUrl inviteUrl;
+	inviteUrl
+	        .setScheme("rs")
+	        .setHost(dts.id.toStdString())
+	        .setQueryKV("name", dts.name);
+
+	if(!bareBones)
+	{
+		if(!dts.dyndns.empty()) inviteUrl.setQueryKV("dyndns", dts.dyndns);
+
+		if(dts.isHiddenNode)
+			inviteUrl.setQueryKV("hiddenNodeAddress", dts.hiddenNodeAddress)
+			        .setQueryKV(
+			            "hiddenNodePort", std::to_string(dts.hiddenNodePort) )
+			        .setQueryKV("hiddenType", std::to_string(dts.hiddenType));
+		else if(dts.extPort && !dts.extAddr.empty())
+			inviteUrl.setQueryKV("extAddr", dts.extAddr)
+			        .setQueryKV("extPort", std::to_string(dts.extPort));
+		else if ( !dts.ipAddressList.empty() &&
+		          !dts.ipAddressList.front().empty() )
+			inviteUrl.setQueryKV("locator", dts.ipAddressList.front());
+
+		if(includeFingerprint && ( !dts.fpr.isNull() ||
+		                           getGPGDetails(dts.gpg_id, dts) ))
+			inviteUrl.setQueryKV("fpr", dts.fpr.toStdString());
+	}
+
+	invite = inviteUrl.toString();
+	if(formatRadix)
+		Radix64::encode(
+		            reinterpret_cast<const uint8_t*>(invite.data()),
+		            static_cast<int>(invite.length()), invite );
+
+	return true;
+}
+
 bool p3Peers::acceptInvite( const std::string& invite,
                             ServicePermissionFlags flags )
 {
 	if(invite.empty()) return false;
+	if(acceptLongInvite(invite, flags)) return true;
+
+	RsUrl url(invite);
```

Comment:
ok, same here. Just remve the url stuff and only keep the invite to be the base64 string.

Reply:
NACK

Reply:
This is really not resolved.

Reply:
yeah but let's keep this discussion only in one place